library/core/src/iter/range.rs

```diff
@@ -3,21 +3,23 @@ use crate::convert::TryFrom;
 use crate::mem;
 use crate::ops::{self, Try};
 
-use super::{FusedIterator, TrustedLen, TrustedRandomAccess};
+use super::{FusedIterator, TrustedLen, TrustedRandomAccess, TrustedStep};
+
+// Safety: All invariants are upheld.
+macro_rules! unsafe_impl_trusted_step {
+    ($($type:ty)*) => {$(
+        #[unstable(feature = "trusted_step", issue = "none")]
+        unsafe impl TrustedStep for $type {}
+    )*};
+}
+unsafe_impl_trusted_step![char i8 i16 i32 i64 i128 isize u8 u16 u32 u64 u128 usize];
 
 /// Objects that have a notion of *successor* and *predecessor* operations.
 ///
 /// The *successor* operation moves towards values that compare greater.
 /// The *predecessor* operation moves towards values that compare lesser.
-///
-/// # Safety
-///
-/// This trait is `unsafe` because its implementation must be correct for
-/// the safety of `unsafe trait TrustedLen` implementations, and the results
-/// of using this trait can otherwise be trusted by `unsafe` code to be correct
-/// and fulfill the listed obligations.
```

Comment:
Can you note the relationship to TrustedStep here (likely just linking as a See also), to connect readers to it? If we stabilize Step first that might want to be dropped, but for now it seems good.

Reply:
I can, but is this really desired? A quick <kbd>ctrl</kbd><kbd>F</kbd> shows not a single instance of `TrustedLen` on the public docs for `Iterator`.

Reply:
Yeah, I guess it's probably not necessary. Seems okay as-is.